include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp

```diff
@@ -0,0 +1,404 @@
+        void configureChannel(const DacOversamplingRatio& osr = DacOversamplingRatio::Osr1
+                            , uint8_t refreshPeriod = 0
+                            , const bool dither = false
+                            , const bool externalFilter = false)
+        {
+          static_assert(DacIndex < 2, "There are only two dac channels available, 0 and 1");
+          const bool shouldRestart = DAC->CTRLA.bit.ENABLE;
+          // disable DAC controller if it is enabled
+          if(shouldRestart) disable();
+          // enable channel
+          enableChannel<DacIndex>();
+          if(refreshPeriod > 15) refreshPeriod = refreshPeriod % 16; //< refresh period must be between 0 and 15
+          uint8_t dacCtrl = DAC->DACCTRL[DacIndex].reg;
+          // set oversampling
+          dacCtrl &= ~DAC_DACCTRL_OSR_Msk;
+          dacCtrl |= static_cast<uint8_t>(osr) << DAC_DACCTRL_OSR_Pos;
+          // refresh period
+          dacCtrl &= ~DAC_DACCTRL_REFRESH_Msk;
+          dacCtrl |= static_cast<uint8_t>(refreshPeriod) << DAC_DACCTRL_REFRESH_Pos;
+          // dither
+          if(dither) dacCtrl |= DAC_DACCTRL_DITHER;
+          else dacCtrl &= ~DAC_DACCTRL_DITHER;
+          // filter
+          if(externalFilter) dacCtrl |= DAC_DACCTRL_FEXT;
+          else dacCtrl &= ~DAC_DACCTRL_FEXT;
+          DAC->DACCTRL[DacIndex].reg = dacCtrl;
+          // reenable DAC controller if it was enabled before
+          if(shouldRestart) enable();
+        }
+
+        /**
+         * @brief Enable specified DAC instance and configure the
+         * pin connected to DAC.
+         * 
+         * @tparam DacIndex The DAC instance (0 or 1)
+         */
+        template<uint8_t DacIndex>
+        void enableChannel()
+        {
+          static_assert(DacIndex < 2, "There are only two dac channels available, 0 and 1");
+          DAC->DACCTRL[DacIndex].reg |= DAC_DACCTRL_ENABLE;
+          // config pins
+          // PA02(DAC0) and PA05(DAC1)
+          if constexpr(DacIndex == 0)
+          {
+            BareCpper::gpioFunction<PA02>(PA02{}, std::optional<ATsamd5x::Peripheral>{ATsamd5x::Peripheral::B});
+          }
+          else
+          {
+            BareCpper::gpioFunction<PA05>(PA05{}, std::optional<ATsamd5x::Peripheral>{ATsamd5x::Peripheral::B});
+          }
+        }
+
+        /**
+         * @brief Disable specified DAC instance and release
+         * the pin associated with the DAC instance.
+         * 
+         * @tparam DacIndex The DAC instance (0 or 1)
+         */
+        template<uint8_t DacIndex>
+        void disableChannel()
+        {
+          static_assert(DacIndex < 2, "There are only two dac channels available, 0 and 1");
+          DAC->DACCTRL[DacIndex].reg &= ~DAC_DACCTRL_ENABLE;
+          // config pins
+          // PA02(DAC0) and PA05(DAC1)
+          // disable pmuxen
+          if constexpr(DacIndex == 0)
+          {
+            BareCpper::gpioFunction<PA02>(PA02{}, std::nullopt);
```

Comment:
PA02 is design specific? Can be solved in future if needs more refactor to remove?

Reply:
The outputs of the DACs are always connected to PA02(DAC0) and PA05(DAC1), there are no other pins on the MCU to be used as DAC outputs.

Reply:
Thanks, that explains this 👍